### PR TITLE
[config] Add option to skip by tensorflow-1 or tensorflow-2

### PR DESCRIPTION
**Pull request by arjkesh**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description
- Saw @junpuf comment in #1395 and realized we didn't have the option to skip by TF1 or TF2. Adding in this option, as it Is useful for iterating on TF1 or TF2, respectively

### Tests
- Tested on 2 revisions in this PR

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/main.py [hunk 1]
```diff
@@ -33,6 +33,16 @@ def main():
     neuron_build_mode = parse_dlc_developer_configs("dev", "neuron_mode")
     graviton_build_mode = parse_dlc_developer_configs("dev", "graviton_mode")
 
+    # Write empty dict to JSON file, so subsequent buildspec steps do not fail in case we skip this build
+    utils.write_to_json_file(constants.TEST_TYPE_IMAGES_PATH, {})
```
Comment (arjkesh) [inline src/main.py:37]: moved this up, since it seemed arbitrarily at bottom of file

File: src/main.py [hunk 2]
```diff
@@ -44,29 +54,27 @@ def main():
         and args.framework not in frameworks_to_skip
     )
     # An EI dedicated builder will work if in EI mode and its framework not been disabled
-    ei_builder_enabled = (
-        ei_dedicated and ei_build_mode and args.framework not in frameworks_to_skip
-    )
+    ei_builder_enabled = ei_dedicated and ei_build_mode and args.framework not in frameworks_to_skip
```
Comment (arjkesh) [inline src/main.py:57]: done automatically by code formatting tool

File: src/main.py [hunk 3]
```diff
@@ -44,29 +54,27 @@ def main():
...
 
     # A NEURON dedicated builder will work if in NEURON mode and its framework has not been disabled
-    neuron_builder_enabled = (
-        neuron_dedicated
-        and neuron_build_mode
-        and args.framework not in frameworks_to_skip
-    )
+    neuron_builder_enabled = neuron_dedicated and neuron_build_mode and args.framework not in frameworks_to_skip
```
Comment (arjkesh) [inline src/main.py:60]: done automatically by code formatting tool

File: src/main.py [hunk 4]
```diff
@@ -44,29 +54,27 @@ def main():
...
 
     # A GRAVITON dedicated builder will work if in GRAVITON mode and its framework has not been disabled
-    graviton_builder_enabled = (
-        graviton_dedicated
-        and graviton_build_mode
-        and args.framework not in frameworks_to_skip
-    )
+    graviton_builder_enabled = graviton_dedicated and graviton_build_mode and args.framework not in frameworks_to_skip
```
Comment (arjkesh) [inline src/main.py:64]: done automatically by code formatting tool

File: src/main.py [hunk 5]
```diff
@@ -44,29 +54,27 @@ def main():
...
 
-    utils.write_to_json_file(constants.TEST_TYPE_IMAGES_PATH, {})
     # A builder will always work if it is in non-PR context
-    if general_builder_enabled or ei_builder_enabled or neuron_builder_enabled or graviton_builder_enabled or build_context != "PR":
+    if (
```
Comment (arjkesh) [inline src/main.py:66]: done automatically by code formatting tool

File: src/main.py [hunk 6]
```diff
@@ -44,29 +54,27 @@ def main():
...
+        general_builder_enabled
+        or ei_builder_enabled
+        or neuron_builder_enabled
+        or graviton_builder_enabled
+        or build_context != "PR"
+    ):
         utils.build_setup(
-            args.framework, device_types=device_types, image_types=image_types, py_versions=py_versions,
+            args.framework,
```
Comment (arjkesh) [inline src/main.py:74]: done automatically by code formatting tool

File: src/main.py [hunk 7]
```diff
@@ -33,6 +33,16 @@ def main():
     neuron_build_mode = parse_dlc_developer_configs("dev", "neuron_mode")
     graviton_build_mode = parse_dlc_developer_configs("dev", "graviton_mode")
 
+    # Write empty dict to JSON file, so subsequent buildspec steps do not fail in case we skip this build
+    utils.write_to_json_file(constants.TEST_TYPE_IMAGES_PATH, {})
+
+    # Add ability to skip tensorflow-1 or tensorflow-2
+    build_arn = utils.get_codebuild_build_arn()
+    if "tensorflow-1" in build_arn and "tensorflow-1" in frameworks_to_skip:
```
Comment (tejaschumbalkar) [inline src/main.py:41]: do we need to check the `build_context` here?

Comment (arjkesh) [reply src/main.py:41]: yeah we can do that

**Review by tejaschumbalkar — APPROVED** [review]

**Pull request merged**